### ISSUE TITLE
lsb_release -a / --version info

### ISSUE THREAD

**Issue by Boostrix** [hdkiller/Auto-GPT-SystemInfo#4]
Howdy, turns out you solved a problem that I just spent an hour fixing.

Just wanting to say, as I mentioned on the Auto-GPT project, I ended up using lsb_release -a (on Linux  obviously) to provide surrounding context, and I use a "preparation" step before running shell commands, to get their version info via the equivalent of <command> --version to get customized commands for the underlying environment.

The idea was prior to executing a shell command to check a local json file which serves as key/value storage, to look up the location(path)/version number of each command and use that info for future invocations.

So the basic idea [on *nix] was:
- lsb_release -a (once during init)
- cat /proc/cpuinfo (once during init)
- cat /proc/meminfo (to show resource utilization at runtime)
- which $TOOL (look up the path/location of the tool), store the unix time or md5hash of the tool, to update the JSON if needed
- $TOOL --version (determine the installed version)

and store the corresponding data inside a JSON file for later use.

More specifically, I figured out that this contextual information is much more useful and effective if it is only conditionally added directly prior to running shell commands, as per: https://github.com/Significant-Gravitas/Auto-GPT/issues/2987#issuecomment-1528954237

Maybe some food for thought ...

Thanks & all the best

**Comment by Emasoft** [comment]
Good idea. This should be automatically executed by all the plugins and commands, no need to create a specific plugin for this. Just the system info plugin make sense but only to get a diagnostic report to the user if he asks the AI. 
Just be sure to make this cross platform (via Python maybe?).

**Comment by Boostrix** [comment]
Yes, I just elaborated on the idea here: https://github.com/Significant-Gravitas/Auto-GPT/issues/2987#issuecomment-1531131136
